docs/docsite/rst/dev_guide/testing_running_locally.rst

```diff
@@ -42,6 +42,14 @@ Use the ``ansible-test shell`` command to get an interactive shell in the same e
 * ``ansible-test shell --docker`` - Open a shell in the default docker container.
 * ``ansible-test shell --tox --python 3.6`` - Open a shell in the Python 3.6 ``tox`` environment.
 
+.. note:: When using ``ansible-test shell --docker`` to run local tests inside
```

Comment:
If you add `ANSIBLE_KEEP_REMOTE_FILES` to `test/runner/tox.ini` it will work with `--tox` too.

Perhaps instead of focusing on this environment variable specifically, the documentation should list which environment variables are passed through, under which circumstances.

Having a generic note about the limitations of using `--docker` (or `--remote`) is probably better than specifically mentioning `ANSIBLE_KEEP_REMOTE_FILES` in relation to those options. The limitations would be:

- Environment variables are not propagated from the host to the container (`--docker`) or remote (`--remote`)
- Only whitelisted variables are exposed to the test environment (mostly via common_environment)
- Only whitelisted variables are exposed when using tox (`--tox`) (whitelist in `test/runner/tox.ini`)
- Most files from tests are not copied to the host from the container (`--docker` or remote `--remote`)

Reply:
@mattclay I like the more general coverage better than the explicit example, I don't know enough about the testing to properly document it though. Would this just be what you've written in the comment above? $other?

Reply:
Basically what I included in my comment. You'll probably want to expand the 2nd and 3rd bullet points to list the specific variables for each.

Reply:
@mattclay I don't know enough about it to really expand, but I made a section for it. Let me know what you think, I'm happy to include any improvements you recommend.